patchwork-extensions-keybinds/src/main/java/com/patchworkmc/mixin/extensions/keybinds/MixinKeyBinding.java

```diff
@@ -0,0 +1,206 @@
+
+import javax.annotation.Nonnull;
+
+import net.minecraftforge.client.extensions.IForgeKeybinding;
+import net.minecraftforge.client.settings.IKeyConflictContext;
+import net.minecraftforge.client.settings.KeyBindingMap;
+import net.minecraftforge.client.settings.KeyConflictContext;
+import net.minecraftforge.client.settings.KeyModifier;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.resource.language.I18n;
+import net.minecraft.client.util.InputUtil;
+
+import com.patchworkmc.impl.extensions.keybinds.IPatchworkKeyBinding;
+
+@Mixin(KeyBinding.class)
+public abstract class MixinKeyBinding implements IForgeKeybinding, IPatchworkKeyBinding {
+	@Shadow
+	private InputUtil.KeyCode keyCode;
+
+	@Mutable
+	@Shadow
+	@Final
+	private static Map<InputUtil.KeyCode, KeyBinding> keysByCode;
+
+	@Shadow
+	public abstract String getCategory();
+
+	@Shadow
+	public abstract String getId();
+
+	@Shadow
+	@Final
+	private static Map<String, Integer> categoryOrderMap;
+	private KeyModifier keyModifierDefault = KeyModifier.NONE;
+	private KeyModifier keyModifier = KeyModifier.NONE;
+	private IKeyConflictContext keyConflictContext = KeyConflictContext.UNIVERSAL;
+
+	@Inject(method = "<clinit>", at = @At("RETURN"))
+	private static void staticInit(CallbackInfo ci) {
+		keysByCode = new KeyBindingMap();
+	}
+
+	@Inject(method = "onKeyPressed", at = @At("HEAD"), cancellable = true)
+	private static void onKeyPressed(InputUtil.KeyCode keyCode, CallbackInfo ci) {
+		ci.cancel();
+
+		for (KeyBinding keybinding : ((KeyBindingMap) keysByCode).lookupAll(keyCode)) {
+			if (keybinding != null) {
+				((KeyBindingHooks) keybinding).setTimesPressed(((KeyBindingHooks) keybinding).getTimesPressed() + 1);
+			}
+		}
+	}
+
+	@Inject(method = "setKeyPressed", at = @At("HEAD"), cancellable = true)
+	private static void onKeyPressed(InputUtil.KeyCode keyCode, boolean pressed, CallbackInfo ci) {
+		ci.cancel();
+
+		for (KeyBinding keybinding : ((KeyBindingMap) keysByCode).lookupAll(keyCode)) {
+			if (keybinding != null) {
+				((KeyBindingHooks) keybinding).setPressed(pressed);
+			}
+		}
+	}
+
+	@Inject(method = "isPressed", at = @At("RETURN"), cancellable = true)
+	private void isPressed(CallbackInfoReturnable<Boolean> cir) {
+		cir.setReturnValue(cir.getReturnValueZ() && getKeyConflictContext().isActive() && getKeyModifier().isActive(getKeyConflictContext()));
+	}
+
+	@Inject(method = "method_1430", at = @At("HEAD"), cancellable = true)
+	private void compareTo(KeyBinding keyBinding, CallbackInfoReturnable<Integer> cir) {
```

Comment:
After looking at the implementation in Forge, it looks like the purpose of this patch is to handle null entries in the categoryOrderMap, no more. This injector is a silent `@Overwrite`, I think this should be rewritten to just fulfill the purpose of handling nulls without clobbering vanilla behavior.
```suggestion
	private void handleNullCategories(KeyBinding keyBinding, CallbackInfoReturnable<Integer> cir) {
```